skypy/halos/_colossus.py

```diff
@@ -203,3 +203,137 @@ def colossus_mf(redshift, model, mdef, m_min, m_max, sky_area, cosmology,
+        to the mass definition, mdef.
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
+    redshift : float
+        Halo redshift
+    model : string
+        The model of the c-M relation which is available in colossus.
+    cosmology : astropy.cosmology.Cosmology
+        Astropy cosmology object
+    sigma8 : float
+        Cosmology parameter, amplitude of the (linear) power spectrum on the
+        scale of 8 Mpc/h.
+    ns : float
+        Cosmology parameter, spectral index of scalar perturbation power spectrum.
+
+    Returns
+    -------
+    concentration : float or array_like
+        Halo concentration(s); has the same dimensions as mass.
+
+    '''
+    from colossus.cosmology.cosmology import fromAstropy
+    from colossus.halo import concentration
+
+    fromAstropy(cosmology, sigma8=sigma8, ns=ns)
+
+    c = concentration.concentration(mass, mdef, redshift, model)
+
+    return c
+
+
+def radius(mass, concentration, redshift, mdef, Delta, cosmology, sigma8, ns):
+    r'''Calculate the scale radius and the spherical overdensity radius of halo by assuming
+    the NFW model.
+
+    This function calculates the scale radius and any spherical overdensity radius
+    for NFW dark matter halo.
+
+    Parameters
+    ----------
+    mass : float
+        A spherical overdensity halo mass in units of solar mass/h, corresponding
+        to the mass definition, mdef.
+    concentration : float
+        The concentration corresponding to the given halo mass and mass definition.
+    redshift : float
+        The halo redshift value.
+    mdef : str
+        Halo mass definition for spherical overdensities used by colossus.
+    Delta : str
+        The mass definition for which the spherical overdensity radius is computed.
+    cosmology : astropy.cosmology.Cosmology
+        Astropy cosmology object
+    sigma8 : float
+        Cosmology parameter, amplitude of the (linear) power spectrum on the
+        scale of 8 Mpc/h.
+    ns : float
+        Cosmology parameter, spectral index of scalar perturbation power spectrum.
+
+    Returns
+    -------
+    rs : float
+        The scale radius in physical kpc/h.
+    RDelta : float
+        Spherical overdensity radius of a given mass definition Delta.
+
+    '''
+    from colossus.cosmology.cosmology import fromAstropy
+    from colossus.halo import profile_nfw
+
+    fromAstropy(cosmology, sigma8=sigma8, ns=ns)
+
+    prof = profile_nfw.NFWProfile(M=mass, c=concentration, z=redshift, mdef=mdef)
+    rs = prof.par['rs']
+    RDelta = prof.RDelta(redshift, Delta)
+
+    return rs, RDelta
+
+
+def Delta_Sigma(mass, concentration, redshift, mdef, radius, cosmology, sigma8, ns):
+    r'''The excess surface density at given radius by assuming the NFW model.
+
+    This function uses Colossus routines to compute the excess surface density profile,
+    which is defined as Delta_Sigma(R) = Sigma(<R) − Sigma(R).
```

Comment:
```suggestion
    r'''Excess surface density 
    
    This function computes the excess surface density at a given radius, assuming the NFW model.
    This function uses Colossus routines to compute the excess surface density profile,
    which is defined as Delta_Sigma(R) = Sigma(<R) − Sigma(R).
```
Or something along those lines.